zio-schema/shared/src/main/scala/zio/schema/Schema.scala

```diff
@@ -432,16 +432,18 @@ object Schema extends SchemaPlatformSpecific with SchemaEquality {
     val transient: Boolean =
       annotations.exists(_.isInstanceOf[transientField])
 
-    val nameAndAliases: scala.collection.immutable.Set[String] =
-      annotations.collect {
-        case aliases: fieldNameAliases => aliases.aliases
-        case f: fieldName              => Seq(f.name)
-      }.flatten.toSet + name
-
     val fieldName: String = annotations.collectFirst {
       case f: fieldName => f.name
     }.getOrElse(name)
 
+    val nameAndAliases: scala.collection.immutable.Set[String] =
+      annotations.foldLeft(scala.collection.immutable.Set(fieldName)) { (acc, annotation) =>
```

Comment:
is there a difference besides making it a fold?

Reply:
It avoids reusing of an old (renamed) field name as an alias